met-web/src/components/metadataManagement/TaxonEditForm.tsx

```diff
@@ -0,0 +1,325 @@
+import {
+    TextField,
+    Switch,
+    FormControlLabel,
+    FormControl,
+    FormGroup,
+    Select,
+    Grid,
+    InputLabel,
+    Button,
+    MenuItem,
+    Tooltip,
+    Avatar,
+    Collapse,
+    Box,
+} from '@mui/material';
+import { Save, Check, Edit, Close, Delete, Error, VerifiedUser, ShieldMoon, Queue, AddBox } from '@mui/icons-material';
+import * as yup from 'yup';
+import React, { useContext, useEffect } from 'react';
+import { MetadataTaxon } from 'models/engagement';
+import { ActionContext } from './ActionContext';
+import { useAppDispatch } from 'hooks';
+import { If, Then, Else } from 'react-if';
+import { TaxonTypes } from './TaxonTypes';
+import { TaxonType } from './types';
+import PresetValuesEditor from './presetFieldsEditor/PresetValuesEditor';
+import { useForm, SubmitHandler, Controller, FormProvider } from 'react-hook-form';
+import { MetHeader3 } from 'components/common';
+import { openNotification } from 'services/notificationService/notificationSlice';
+
+const TaxonEditForm = ({ taxon }: { taxon: MetadataTaxon }): JSX.Element => {
+    const { setSelectedTaxonId, updateMetadataTaxon, removeMetadataTaxon } = useContext(ActionContext);
+    const dispatch = useAppDispatch();
+    const methods = useForm<MetadataTaxon>({
+        defaultValues: {
+            name: taxon.name,
+            description: taxon.description ?? '',
+            freeform: taxon.freeform,
+            one_per_engagement: taxon.one_per_engagement,
+            data_type: taxon.data_type,
+            preset_values: taxon.preset_values,
+        },
+    });
+    const {
+        handleSubmit,
+        control,
+        watch,
+        formState: { isDirty },
+        reset,
+        setError,
+    } = methods;
+
+    useEffect(() => {
+        reset({
+            name: taxon.name ?? '',
+            description: taxon.description ?? '',
+            freeform: taxon.freeform ?? false,
+            one_per_engagement: taxon.one_per_engagement ?? false,
+            data_type: taxon.data_type ?? 'text',
+            preset_values: taxon.preset_values ?? [],
+        });
+    }, [taxon, reset]);
+
+    // Watch data_type to update taxonType
+    const dataType = watch('data_type');
+    const taxonType = TaxonTypes[dataType as keyof typeof TaxonTypes];
+    // Watch freeform to update label
+    const isFreeform = watch('freeform');
+    const isMulti = !watch('one_per_engagement');
+    const presetValues = watch('preset_values');
+
+    const schema = yup.object().shape({
+        name: yup.string().required('Name is required').max(64, 'Name is too long!'),
+        description: yup.string().max(255, 'Description is too long!').strip(),
```

Comment:
Will this error message appear in real time as the user surpasses the limit? If not, we may want to include the max length in the error message